gum/backend-windows/gumtls-windows.c

```diff
@@ -28,6 +53,163 @@ gum_tls_key_free (GumTlsKey key)
   TlsFree (key);
 }
 
+
+void
+_gum_tls_init (void)
+{
+#if defined (HAVE_I386)
+  gum_spinlock_init (&_gum_tls_tmp_keys_lock);
+  memset (_gum_tls_tmp_keys, 0, sizeof (_gum_tls_tmp_keys));
+#else
+  (void);
+#endif
+}
+
+#if defined (HAVE_I386)
+
+static
+gpointer
+_gum_tls_key_get_tmp_value (GumTlsKey key)
+{
+  guint i;
+  GumThreadId tid = gum_process_get_current_thread_id ();
+  gpointer value = NULL;
+  gum_spinlock_acquire (&_gum_tls_tmp_keys_lock);
+  for (i = 0; i < MAX_TMP_TLS_KEY; i++)
+  {
+    if ((_gum_tls_tmp_keys[i].tid == tid) && (_gum_tls_tmp_keys[i].key == key))
+    {
+      value = _gum_tls_tmp_keys[i].value;
+      break;
+    }
+  }
+  g_assert (i < MAX_TMP_TLS_KEY);
+  gum_spinlock_release (&_gum_tls_tmp_keys_lock);
+  return NULL;
```

Comment:
Always return `NULL`? Also this function would be easier to read if it had some declarations moved up from the statements, and a few empty lines to make it breathe a bit. :)